test/externalTests/common.sh

```diff
@@ -104,7 +104,7 @@ function replace_version_pragmas
     # Replace fixed-version pragmas (part of Consensys best practice).
     # Include all directories to also cover node dependencies.
     printLog "Replacing fixed-version pragmas..."
-    find . test -name '*.sol' -type f -print0 | xargs -0 sed -i -e 's/pragma solidity [\^0-9\.]*/pragma solidity >=0.0/'
+    find . test -name '*.sol' -type f -print0 | xargs -0 sed -i -E 's/pragma solidity [^;]+;/pragma solidity >=0.0;/'
```

Comment:
What is `-e` vs `-E` ?

Reply:
- `-e` just says that the `'s/pragma solidity ...'` part is the sed script to execute.
- `-E` is `--regexp-extended`. It's nice because it allows not escaping `+`, `(` and several other symbols. I also had better luck writing regexes that reliably work on macOS that way.

Actually it should have been `-E -e`. I keep forgetting about it because it works on my system even without `-e`, probably because it's the default command. But this might not be the case with BSD sed.

Reply:
`-E` seems to be present on macos too:
```
     -E      Interpret regular expressions as extended (modern) regular expressions rather than basic regular expressions (BRE's).  The re_format(7)
             manual page fully describes both formats.
```

Is this script used in macos tests on CI?

Reply:
It's not (the bare `-i` is a giveaway that it would not work on mac) but I'm inclined to use it by default because without it you need to escape a lot of stuff making the regexes even harder to read than they usually are. The fact that it works more consistently between Linux and macOS than the default mode is just an added bonus :)

Reply:
Can you create a tracking issue to run all tests on macos CI too?

Reply:
Sure. What about Windows and Linux? Currently these external tests only run on `soljson` via Truffle.

Reply:
Done: #10446.